six/modules/c++/cphd/source/PVPBlock.cpp

```diff
@@ -288,32 +288,31 @@ void PVPBlock::PVPSet::read(const Pvp& p, sys::ubyte* dest) const
 /*
  * Initialize PVP Array with a data object
  */
-PVPBlock::PVPBlock(const Data& d, const Pvp& p)
+PVPBlock::PVPBlock(const Pvp& p, Data& d)
 {
     mPvp = p;
-    mNumBytesPerVector = mPvp.getReqSetSize()*sizeof(double);
+    mNumBytesPerVector = d.getNumBytesPVPSet();
     mData.resize(d.getNumChannels());
     for (size_t ii = 0; ii < d.getNumChannels(); ++ii)
     {
         mData[ii].resize(d.getNumVectors(ii));
     }
-    size_t calculateBytesPerVector = getNumBytesPVPSet();
+    size_t calculateBytesPerVector = mPvp.getReqSetSize()*sizeof(double);
     if (six::Init::isUndefined<size_t>(mNumBytesPerVector) ||
         calculateBytesPerVector > mNumBytesPerVector)
     {
         mNumBytesPerVector = calculateBytesPerVector;
+        d.numBytesPVP = calculateBytesPerVector;
     }
-
 }
 
 /*
  * Initialize PVP Array with custom parameters
  */
-PVPBlock::PVPBlock(size_t numBytesPerVector,
-                   size_t numChannels,
+PVPBlock::PVPBlock(size_t numChannels,
```

Comment:
As this constructor is not passed a Data object, the numBytesPVP metadata can't be updated. I mentioned that in the comments.
I included this constructor to match the interface of VBM.h but not sure how useful this is.